api/src/ballots.py

```diff
@@ -18,8 +18,11 @@
 def ballots():
  
   # required parameters - throws error if not present
-  state = req.args['state'].upper()
-  elec_dt = datetime.strptime(req.args['election_dt'], '%m-%d-%Y')
+  try:
+    state = req.args['state'].upper()
+    elec_dt = datetime.strptime(req.args['election_dt'], '%m-%d-%Y')
+  except:
+    return "404 bad request"
```

Comment:
Better way to do error returns: https://flask.palletsprojects.com/en/1.1.x/patterns/errorpages/#returning-api-errors-as-json. Change all the returns to follow that pattern and this should be good